### PR TITLE
[no sq] Small network-related changes

### PR DESCRIPTION
**Pull request by sfan5**
minor stuff. won't fix any of our current problems

## How to test

Check that you can still join singleplayer without noticable delay.

### CODE REVIEW

File: src/network/connectionthreads.cpp [hunk 1]
```diff
@@ -59,14 +59,24 @@ static inline u8 readChannel(const u8 *packetdata)
...
 	m_max_packet_size(max_packet_size),
 	m_timeout(timeout),
-	m_max_data_packets_per_iteration(g_settings->getU16("max_packets_per_iteration"))
+	m_max_data_packets_per_iteration(g_settings->getU16(MPPI_SETTING))
 {
-	SANITY_CHECK(m_max_data_packets_per_iteration > 1);
+	auto &mppi = m_max_data_packets_per_iteration;
+	const auto mppi_default = Settings::getLayer(SL_DEFAULTS)->getU16(MPPI_SETTING);
+	if (mppi != mppi_default) {
+		warningstream << "You are running the network code with a non-default "
```
Comment (appgurueu) [inline src/network/connectionthreads.cpp:74]: Perhaps also update the setting description to include this warning so people don't change it in the first place?

**Review by appgurueu — APPROVED** [review]
LGTM

---

BTW after this PR, only the sending of player pos etc. remains unreliable, we might want to consider changing that.

File: src/network/clientopcodes.cpp [hunk 2]
```diff
@@ -144,7 +144,7 @@ const ServerCommandFactory serverCommandFactoryTable[TOSERVER_NUM_MSG_TYPES] =
 {
 	null_command_factory, // 0x00
 	null_command_factory, // 0x01
-	{ "TOSERVER_INIT",               1, false }, // 0x02
+	{ "TOSERVER_INIT",               1, true }, // 0x02
```
Comment (nerzhul) [inline src/network/clientopcodes.cpp:147]: is this retrocompatible ? i'm not sure

Comment (sfan5) [reply src/network/clientopcodes.cpp:147]: yes, the server does not care

**Comment by red-001** [comment]
Was that packet not sent as unreliable for some good reason? I'll admit I presumed there was a reason for it.

**Comment by sfan5** [comment]
0.3 re-sent TOSERVER_INIT with a manual timer:
https://github.com/minetest/minetest/blob/1a3d25c5ad72e9c0096601dbff9c06512ae3f2bf/src/client.cpp#L404-L417
I assume it was the first packet at the time, instead of the empty packet to open a connection that we have now.

**Comment by sfan5** [comment]
uhhh nevermind we still do this:
https://github.com/minetest/minetest/blob/b0107144261f6513bbde95323c9663a71ed8f3c9/src/client/client.cpp#L448-L462

**Review by SmallJoker — APPROVED** [review]
:ballot_box_with_check: Warning for non-default `max_packets_per_iteration` value
:ballot_box_with_check: Joined a few public servers. Works.
:ballot_box_with_check: Local server + join works.

**Pull request merged**